builder/build.go

```diff
@@ -667,6 +667,16 @@ func Build(pkgName, outpath, tmpdir string, config *compileopts.Config) (BuildRe
 		ldflags = append(ldflags, "--no-entry")
 	}
 
+	if config.Options.BuildMode == "wasi-legacy" {
+		if !strings.HasPrefix(config.Triple(), "wasm32-") {
+			return result, fmt.Errorf("buildmode wasi-legacy is only supported on wasm")
+		}
+
+		if config.Options.Scheduler != "none" {
+			return result, fmt.Errorf("buildmode wasi-legacy only supports scheduler=none")
+		}
```

Comment:
Slightly better would be:

```go
if config.Scheduler() != "none"
```

The reason is that for example `-target=wasm-unknown` already defaults to scheduler "none", so the explicit command line flag (found via `config.Options.Scheduler`) isn't needed.